I couldn't get the warning because Python applies `-O3`:

```
/opt/bin/gcc -c -Wall -IInclude -I. Python/pystrtod.c
Python/pystrtod.c: In function ‘format_float_short’:
Python/pystrtod.c:1063:13: warning: ‘strncpy’ output truncated before terminating nul copying 3 bytes from a string of the same length [-Wstringop-truncation]
             strncpy(p, "ERR", 3);
             ^~~~~~~~~~~~~~~~~~~~
/opt/bin/gcc -c -O3 -Wall -IInclude -I. Python/pystrtod.c
```

--- a/Python/pystrtod.c
+++ b/Python/pystrtod.c
@@ -1060,8 +1060,8 @@ format_float_short(double d, char format_code,
         else {
             /* shouldn't get here: Gay's code should always return
                something starting with a digit, an 'I',  or 'N' */
-            strncpy(p, "ERR", 3);
-            /* p += 3; */
+            /* strncpy(p, "ERR", 3);
+               p += 3; */
             Py_UNREACHABLE();
         }
         goto exit;

